sqldelight-runtime/src/main/java/com/squareup/sqldelight/db/SqlDatabaseHelper.kt

```diff
@@ -13,17 +13,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.squareup.sqldelight;
+package com.squareup.sqldelight.db
 
-import android.database.Cursor;
-import android.support.annotation.CheckResult;
-import android.support.annotation.NonNull;
+interface SqlDatabaseHelper {
+  fun getConnection(): SqlDatabaseConnection
+  fun close()
 
-/** Creates instances of {@code T} from rows in a {@link Cursor}. */
-public interface RowMapper<T> {
-  /**
-   * Return an instance of {@code T} corresponding to the values of the current positioned row of
-   * {@code cursor}.
-   */
-  @CheckResult @NonNull T map(@NonNull Cursor cursor);
+  interface Callback {
+    fun onCreate(connection: SqlDatabaseConnection)
```

Comment:
this is just the initial schema being applied?

Reply:
I wonder if it’s actually easier to have one onSchemaUpdate() method with an oldVersion that can be null?